lib/media/track/mediatrack.js

```diff
@@ -123,10 +123,17 @@ class MediaTrack extends Track {
     this._log.debug('Initializing');
     this._dummyEl = this._createElement();
 
-    this.mediaStreamTrack.addEventListener('ended', function onended() {
-      self._end();
-      self.mediaStreamTrack.removeEventListener('ended', onended);
-    });
+    if (this.mediaStreamTrack && this.mediaStreamTrack.addEventListener) {
+      this.mediaStreamTrack.addEventListener('ended', function onended() {
+        self._end();
+        self.mediaStreamTrack.removeEventListener('ended', onended);
+      });
+    } else if (this.mediaStreamTrack && this.mediaStreamTrack.onended) {
```

Comment:
Should we just do `} else if (this.mediaStreamTrack) {`? 
I thought `this.mediaStreamTrack.onended` is going to be null in this case.

Reply:
You are right! I will adjust it